code/modules/mob/living/carbon/human/life.dm

```diff
@@ -62,6 +62,8 @@
 
 	//No need to update all of these procs if the guy is dead.
 	fall() //VORESTATION EDIT. Prevents people from floating
+	if(unnaturally_resized)

```

Comment:
I'd highly recommend at this point if you're hardcoding this to have this triggered on /Enter and /Exit of the target dorms
Life tick is awful for these things, there's already a ton of stuff piled into Life that shouldn't be there.